### DISCUSSION TITLE
GitHub Discussions

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#5403] [General]
As you can see we've opted in to GitHub's new [discussions](https://github.blog/2020-05-06-new-from-satellite-2020-github-codespaces-github-discussions-securing-code-in-private-repositories-and-more/#discussions) feature, currently in beta. Discussions provide a forum for general discussion separate from the project's development workflow. We believe this has several benefits:

* **Discussions happen close to the code.** Discussions are a part of the GitHub project, and therefore easy for new users to find. And being part of the same platform allows us to easily cross-reference issues and discussions. This has always been a bit of a pain point with the mailing list.

* **Allows issues to remain focused on development.** Despite clear guidance that GitHub issues should be opened only for a select few purposes (e.g. feature requests, bug reports, etc.), we continually see people ignoring the issue templates and asking general questions, in which case the issue is closed and the author is directed to the mailing list. The addition of discussions should help alleviate this.

* **Long-term retention.** A major drawback of Slack discussions is that the history doesn't last beyond a few weeks or months, and it isn't easily searchable. Mailing list discussions are archived indefinitely, but they can be difficult to search. GitHub discussions promise a much more structured and organized platform for communication, and can be filtered and searched very similarly to issues, making it easy to find questions that have already been posed.

Bear in mind that this is GitHub feature is still in beta, so we might see some changes. However, the platform appears promising and I encourage everyone to start making use of it today. Let us know what's on your mind!

**Comment by candlerb** [comment]
Is the aim ultimately to phase out google groups in favour of github discussions?  If not, there will be two places that have to be watched.

**Reply by jeremystretch** [reply]
IMO that would be best, but it's for the community to decide. I share your concern regarding the partitioning of communication.

**Comment by markkuleinio** [comment]
Is there a way to see which topics are new or have new comments since I last visited here?

**Comment by candlerb** [comment]
I notice that `Watch > Custom > Discussions` is now available.  I'm going to try it, although if it spams me for every comment that might not be what I want.
